### PR TITLE
Add a helm chart

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
Push the image to docker hub `oamdev/core-resource-controller` and add a helm chart

### CODE REVIEW

**Comment by ryanzhang-oss** [comment]
https://github.com/crossplane/addon-oam-kubernetes-local/issues/3

File: charts/oam-core-resources/templates/controller-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      securityContext:
+          {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Release.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          command:
+            - /manager
+          args:
+            - "--metrics-addr=127.0.0.1:8080"
```
Comment (wonderflow) [inline charts/oam-core-resources/templates/controller-deployment.yaml]:
```suggestion
            - "--metrics-addr=:8080"
```

File: charts/oam-core-resources/templates/controller-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            - "--enable-leader-election"
+          image: {{ .Values.image.repository }}:{{ .Chart.AppVersion }}
```
Comment (wonderflow) [inline charts/oam-core-resources/templates/controller-deployment.yaml]: we usually don't use `.Chart.AppVersion` as a image tag. You could define a tag in values file like `.Values.image.tag`

Comment (hongchaodeng) [reply charts/oam-core-resources/templates/controller-deployment.yaml]: Another easier way: just make it:
```
image: {{ .Values.image.image }}
```

File: README.md [hunk 3]
```diff
@@ -47,9 +54,16 @@ kubectl apply -f config/samples/sample_application_config.yaml
 ```
 
 *. Verify that the application is running
```
Comment (hongchaodeng) [inline README.md]:
```suggestion
* Verify that the application is running
```

File: README.md [hunk 4]
```diff
@@ -3,6 +3,13 @@
...
  OAM v1alpha2 spec and run it on a kubernetes cluster.  
 
+## Prerequisites
+
+1. go version 1.12+
+2. Kubernetes version v1.15+ with KubeCtl configured
+3. Helm 3.0+ 
+
+
 ## Examples
```
Comment (hongchaodeng) [inline README.md]: Getting started

File: config/samples/core_v1alpha2_manualscalertrait.yaml [hunk 5]
```diff
@@ -3,9 +3,9 @@ kind: ManualScalerTrait
 metadata:
   name: manualscalertrait-sample
 spec:
-  replicaCount: 5
+  replicaCount: 15
```
Comment (hongchaodeng) [inline config/samples/core_v1alpha2_manualscalertrait.yaml:6]: An example with 15 replica is too large.
Make it 1?

File: README.md [hunk 6]
```diff
@@ -47,9 +54,16 @@ kubectl apply -f config/samples/sample_application_config.yaml
 ```
 
 *. Verify that the application is running
+You should see a deployment looking like below
+```
+kubectl get deployments
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+example-appconfig-workload-deployment   10/10   10           10          8m11s
```
Comment (hongchaodeng) [inline README.md:61]: Also update the replica count to the number in manual-scaler trait

File: config/samples/core_v1alpha2_manualscalertrait.yaml (hunk 5 above)
Comment (ryanzhang-oss) [reply config/samples/core_v1alpha2_manualscalertrait.yaml:6]: This is to test the mutation webhook

Comment (wonderflow) [reply config/samples/core_v1alpha2_manualscalertrait.yaml:6]: make it 1~3 for test

**Review by wonderflow — APPROVED** [review]

**Review by hongchaodeng — APPROVED** [review]
lgtm

**Pull request merged**